src/com/intellij/idea/plugin/hybris/impex/assistance/ImpexColumnBreadcrumbsProvider.kt

```diff
@@ -0,0 +1,125 @@
+/*
+ * This file is part of "hybris integration" plugin for Intellij IDEA.
+ * Copyright (C) 2019 EPAM Systems <hybrisideaplugin@epam.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.intellij.idea.plugin.hybris.impex.assistance;
+
+import com.intellij.idea.plugin.hybris.impex.ImpexLanguage
+import com.intellij.idea.plugin.hybris.impex.psi.*
+import com.intellij.idea.plugin.hybris.impex.utils.ImpexPsiUtils
+import com.intellij.lang.Language
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiWhiteSpace
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.ui.breadcrumbs.BreadcrumbsProvider
+import com.intellij.util.ObjectUtils
+
+class ImpexColumnBreadcrumbsProvider : BreadcrumbsProvider {
+
+    override fun getLanguages(): Array<Language> = arrayOf(ImpexLanguage.getInstance())
+
+    override fun acceptElement(element: PsiElement) = element is ImpexFullHeaderParameter
+        || element is ImpexFullHeaderType
+        || element is ImpexAnyHeaderMode
+
+    override fun getElementInfo(psi: PsiElement): String {
+        val headerParameter = getLinkedHeaderParameter(psi)
+        if (headerParameter != null) return headerParameter.text
+
+        val adjustedPsi = adjustWhiteSpaceAndSeparator(psi)
+
+        when (val parent = PsiTreeUtil.getParentOfType(
```

Comment:
extract `val parent...` from the `when()` into standalone variable - right now it is hard to read/understand